projects/extension/build.py

```diff
@@ -116,13 +186,33 @@ def incremental_sql_files() -> list[Path]:
 def check_incremental_sql_files(paths: list[Path]) -> None:
     prev = 0
     for path in paths:
-        this = int(path.name[0:3])
-        if this != prev + 1:
+        check_sql_file_name(path)
+        this = file_number(path)
```

Comment:
WDYT about simplifying this to have only one function that uses a little modification on your regex to extract the both match and extract the file number?

Something like 
```python
def extract_sql_file_number(path: Path) -> int:
    pattern = r"^(\d{3})-[a-z][a-z_-]*\.sql$"
    matches = re.match(pattern, path.name, re.IGNORECASE) # I guess we want to be non case sensitive

    if not matches:
        print(
            f"{path} file name does not match the pattern {pattern}",
            file=sys.stderr,
        )
        sys.exit(1)
    return matches.group(1)
```

Reply:
done